Riot/Modules/Room/Members/Detail/RoomMemberDetailsViewController.m

```diff
@@ -403,37 +438,132 @@ - (void)updateMemberInfo
         }
         
         // Retrieve member's devices
-        NSString *userId = self.mxRoomMember.userId;
         __weak typeof(self) weakSelf = self;
+        
+        if (!RiotSettings.shared.enableCrossSigning)
+        {
+            [self.mxRoom.mxSession.crypto downloadKeys:@[userId] forceDownload:NO success:^(MXUsersDevicesMap<MXDeviceInfo *> *usersDevicesInfoMap, NSDictionary<NSString *,MXCrossSigningInfo *> *crossSigningKeysMap) {
+                
+                if (weakSelf)
+                {
+                    // Restore the status bar
+                    typeof(self) self = weakSelf;
+                    self->devicesArray = usersDevicesInfoMap.map[userId].allValues;
+                    // Reload the full table to take into account a potential change on a device status.
+                    [super updateMemberInfo];
+                }
+                
+            } failure:^(NSError *error) {
+                
+                NSLog(@"[RoomMemberDetailsVC] Crypto failed to download device info for user: %@", userId);
+                if (weakSelf)
+                {
+                    // Restore the status bar
+                    typeof(self) self = weakSelf;
+                    // Notify the end user
+                    NSString *myUserId = self.mainSession.myUser.userId;
+                    [[NSNotificationCenter defaultCenter] postNotificationName:kMXKErrorNotification object:error userInfo:myUserId ? @{kMXKErrorUserIdKey: myUserId} : nil];
+                }
+                
+            }];
+        }
+    }
+    
+    // Complete data update and reload table view
+    [super updateMemberInfo];
+}
 
-        [self.mxRoom.mxSession.crypto downloadKeys:@[userId] forceDownload:NO success:^(MXUsersDevicesMap<MXDeviceInfo *> *usersDevicesInfoMap, NSDictionary<NSString *,MXCrossSigningInfo *> *crossSigningKeysMap) {
-
-            if (weakSelf)
+- (void)refreshUserEncryptionTrustLevel
+{
+    NSString *userId = self.mxRoomMember.userId;
+    
+    if (!userId)
+    {
+        return;
+    }
+    
+    if (self.mxRoom.summary.isEncrypted && self.mxRoom.mxSession.crypto)
+    {
+        [self.mxRoom.mxSession.crypto trustLevelSummaryForUserIds:@[userId] success:^(MXUsersTrustLevelSummary *usersTrustLevelSummary) {
```

Comment:
We should use the sync version of `trustLevelSummaryForUserIds`. It can be used just after the call of `downloadKeys` just above.